server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Enqueuer.java

```diff
@@ -76,6 +76,14 @@ void enQueue(Mail mail) throws MailQueue.MailQueueException {
             .block();
     }
 
+    Mono<Void> reQueue(Mail mail) throws MailQueue.MailQueueException {
+        EnqueueId enqueueId = EnqueueId.generate();
```

Comment:
I would expect we reuse the alredy existing enqueue id. As from a mailqueue api pointof view this is not a newly enqueued email.

It has impact on deletion before / after requeue for instance.